File: app/src/main/java/com/sasaj/graphics/drawingapp/DrawingApplication.java
package com.sasaj.graphics.drawingapp;

import android.app.Application;
import android.content.Context;
import android.os.Handler;

import com.squareup.leakcanary.LeakCanary;

/**
 * Created by DS on 4/22/2017.
 */

public class DrawingApplication extends Application {


    private static Context appContext;
    private static Handler applicationHandler;

    @Override
    public void onCreate() {
        super.onCreate();
        if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return;
        }
        LeakCanary.install(this);

        appContext = getApplicationContext();
        applicationHandler = new Handler(appContext.getMainLooper());
    }

    public static Context getContext() {
        return appContext;
    }

    public static Handler getApplicationHandler() {
        return applicationHandler;
    }


}

File: app/src/main/java/com/sasaj/graphics/drawingapp/DrawingApplication.kt
package com.sasaj.graphics.drawingapp

import android.app.Application
import android.util.Log
import com.sasaj.graphics.drawingapp.di.ApplicationModule
import com.sasaj.graphics.drawingapp.di.DaggerViewModelInjector
import com.sasaj.graphics.drawingapp.di.ViewModelInjector
import io.reactivex.plugins.RxJavaPlugins



/**
 * Created by sjugurdzija on 4/22/2017.
 */

class DrawingApplication : Application(){
    companion object {
        lateinit var injector: ViewModelInjector
    }

    override fun onCreate() {
        super.onCreate()
        RxJavaPlugins.setErrorHandler { e -> Log.e("App", e.message, e)}
        injector = DaggerViewModelInjector.builder()
                .applicationModule(ApplicationModule(applicationContext))
                .build()
    }
}
